org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTDelegateCommandHandler.java

```diff
@@ -70,6 +70,23 @@ public Object executeCommand(String commandId, List<Object> arguments, IProgress
 						// workspaceEdit on the custom command.
 						return result;
 					}
+				case "java.apply.workspaceEdit":
+					final WorkspaceEdit r = (WorkspaceEdit) JSONUtility.toLsp4jModel(arguments.get(0), WorkspaceEdit.class);
+		
+					if (JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences()
+							.isWorkspaceApplyEditSupported()) {
+						JavaLanguageServerPlugin.getInstance().getClientConnection()
+								.applyWorkspaceEdit((WorkspaceEdit) r);

```

Comment:
I don't think you need the cast to `WorkspaceEdit` here. It can just be `.applyWorkspaceEdit(r);` . `r` is of type `WorkspaceEdit`.

Reply:
ditto